corehq/apps/data_dictionary/util.py

```diff
@@ -151,6 +151,25 @@ def get_case_property_label_dict(domain):
     return labels_dict
 
 
+def get_case_property_deprecated_dict(domain):
+    """
+    This returns a dictionary of the structure
+    {
+        case_type: {
+            case_property: is_deprecated,
+            ...
+        },
+        ...
+    }
+    for each case type and case property in the domain
+    """
```

Comment:
> Thinking if we should consider just retaining a list of deprecated properties for each case type instead.

Do you mean instead of having the format of:
```
{
    case_type: {
        case_prop: is_deprecated,
        ...
    }
    ...
}
```

we instead do this:
```
{
    case_type: [
        case_prop
        ...
    ]
    ...
}
```

My only concern with the above approach is that while it is less data sent to the browser, it might be slower on the front-end to identify whether a case property is deprecated as we'll have to iterate through a list instead of directly looking up the case property through a dictionary.

> Additionally, in app manager, I believe we usually deal with one case type at a time? Though not sure if we should worry about that here.

This is true. We send what the active case type is [here](https://github.com/dimagi/commcare-hq/blob/cdd228d7871e83ec7c90bf8a5cd80eb1c97d5f28/corehq/apps/app_manager/views/forms.py#L789), so I'm wondering if, instead of changing the format from a dict to a list, we just send the data for the active case type.

Reply:
> it might be slower on the front-end to identify whether a case property is deprecated as we'll have to iterate through a list instead of directly looking up the case property through a dictionary.

Absolutely. I was thinking about the same as well & was thinking that that change won't be noticeable. If any case type that has lots of case properties, if list lookup makes considerable difference, you could retain the dictionary.

> we just send the data for the active case type

Okay. I was not sure that we won't need other case types on that page. I was wondering if any related cases are referred?

Anyhow, this is not a blocking comment. I was just thinking about different ways here and thought a list would be more efficient. I leave the final call to you.

Reply:
[5cc38da](https://github.com/dimagi/commcare-hq/pull/34425/commits/5cc38daba458a30b51eb79571eda701ed9ab0e01)

Reply:
> Since other data, such as the case property descriptions, are sending data for each case type I'll rather be safe and keep it the same for deprecated case properties

Ah, good detail :+1: 

Thanks!